core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -81,61 +96,32 @@ public boolean validate() {
     return true;
   }
 
-  private ProcessExecutionResult getResultFromProcess(String[] args) {
-    try {
-      Process process = Runtime.getRuntime().exec(args);
-      StringBuilder outputSb = new StringBuilder();
-      try (BufferedReader processOutputReader = new BufferedReader(
-          new InputStreamReader(process.getInputStream()))) {
-        String line;
-        while ((line = processOutputReader.readLine()) != null) {
-          outputSb.append(line);
-          outputSb.append(LINE_SEPARATOR);
-        }
-      }
-      StringBuilder errorSb = new StringBuilder();
-      try (BufferedReader processErrorReader = new BufferedReader(
-          new InputStreamReader(process.getErrorStream()))) {
-        String line;
-        while ((line = processErrorReader.readLine()) != null) {
-          errorSb.append(line);
-          errorSb.append(LINE_SEPARATOR);
-        }
-      }
-      process.waitFor();
-      return new ProcessExecutionResult(process.exitValue(), outputSb.toString().trim(),
-          errorSb.toString().trim());
-    } catch (IOException e) {
-      System.err.println("Failed to execute command.");
-      return new ProcessExecutionResult(1, "", "");
-    } catch (InterruptedException e) {
-      System.err.println("Interrupted.");
-      Thread.currentThread().interrupt();
-      return new ProcessExecutionResult(1, "", "");
+  private boolean validateHdfsSettingParity(Map<String, String> optionsMap) {
+    String serverHadoopConfDirPath;
+    if (optionsMap.containsKey(ValidateEnv.HADOOP_CONF_DIR_OPTION.getOpt())) {
+      serverHadoopConfDirPath = optionsMap.get(ValidateEnv.HADOOP_CONF_DIR_OPTION.getOpt());
+    } else {
+      serverHadoopConfDirPath = System.getenv(HADOOP_CONF_DIR_ENV_VAR);
     }
-  }
-
-  static class ProcessExecutionResult {
-    private final int mExitValue;
-    private final String mOutput;
-    private final String mError;
-
-    public ProcessExecutionResult(int val, String output, String error) {
-      mExitValue = val;
-      mOutput = output;
-      mError = error;
-    }
-
-    public int getExitValue() {
-      return mExitValue;
-    }
-
-    public String getOutput() {
-      return mOutput;
+    if (serverHadoopConfDirPath == null) {
+      System.out.println("Path to server-side hadoop configuration unspecified,"
+          + " aborting validation for HDFS properties.");
+      return false;
```

Comment:
We should not fail the validation just because user does not give configuration file path. User might not have hdfs configured at all and it is an acceptable case.

Reply:
Agree, but I think we still need to give warnings.

Reply:
Sounds good. Let's print a warning message saying the validation is skipped since the configuration file is not specified. The function should return true so users who don't have HDFS configured can still pass the check.

Reply:
Done.